neurovault/apps/statmaps/views.py

```diff
@@ -1202,16 +1202,17 @@ def get_initial_queryset(self):
 
 
 def download_collection(request, cid):
-    from django.http import StreamingHttpResponse
-
     # Files (local path) to put in the .zip
     collection = get_collection(cid, request)
-    filenames = [img.file.path for img in collection.basecollectionitem_set.all()]
+
+    filenames = [img.zip_file.path for img in collection.basecollectionitem_set.instance_of(NIDMResults)] + \
+                [img.file.path for img in
+                 (collection.basecollectionitem_set.instance_of(Image).exclude().instance_of(NIDMResultStatisticMap))]
```

Comment:
The empty `.exclude()` does not do anything. You should use `not_instance_of(NIDMResultStatisticMap)` instead. Coudl you add a test that downloads a collection with NIDMResults as well as normal StatisticMaps (not coming from a NIDM Results object) and makes sure all of the expected files are present in the downloaded zip?